minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Mixins/MixinRandomSkinTexture.java

```diff
@@ -3,24 +3,32 @@
 package com.microsoft.Malmo.Mixins;
 
 
-import com.microsoft.Malmo.Utils.SeedHelper;
-import net.minecraft.client.resources.DefaultPlayerSkin;
+import java.util.Random;
+import java.util.UUID;
+
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
-import java.util.Random;
-import java.util.UUID;
+import net.minecraft.client.resources.DefaultPlayerSkin;
 
 
 @Mixin(DefaultPlayerSkin.class)
 public abstract class MixinRandomSkinTexture {
     // Randomize the skin for agents ignoring UUID
+    private static Integer skinSeed = null;
     @Inject(method = "isSlimSkin", at = @At("HEAD"), cancellable = true)
     private static void isSlimSkin(UUID playerUUID, CallbackInfoReturnable<Boolean> cir){
-        Random rand = new Random();
-        cir.setReturnValue(rand.nextBoolean());
+        if (skinSeed == null) {
+            // TODO now this is not blinking, and randomized, but not consistent between
+            // clients. Ideally, we should somehow sync this between clients -
+            // is there any pseudo-random number in the minecraft world we
```

Comment:
We have UUID of the agent, we could just salt the UUID with some server wide information like map seed?

Reply:
yeah, using map seed sounds great!